code/src/main/kotlin/com/expediagroup/sdk/v2/core/extension/io/InputStreamExtension.kt

```diff
@@ -0,0 +1,21 @@
+package com.expediagroup.sdk.v2.core.extension.io
+
+import okio.Buffer
+import java.io.InputStream
+
+fun InputStream.toBuffer(contentLength: Int, exhaustStream: Boolean = true): Buffer {
+    val buffer = Buffer()
+
+    synchronized(this) {
+
+        if (!exhaustStream) {
+            this.mark(contentLength)
```

Comment:
I read that mark isn't supported in all `InputStream` implementations, therefore, `InputStream` has `markSupported()` method. Do we have concerns about this here?

Reply:
Fixed.